java/java.nativeimage.debugger/src/org/netbeans/modules/java/nativeimage/debugger/actions/DebugNativeImageAction.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.java.nativeimage.debugger.actions;
+
+import org.netbeans.modules.java.nativeimage.debugger.api.NIDebugRunner;
+import java.io.File;
+import java.util.Collections;
+import javax.swing.Action;
+
+import org.netbeans.api.project.Project;
+import org.netbeans.api.project.ProjectUtils;
+import org.netbeans.modules.nativeimage.api.NIUtils;
+import org.netbeans.modules.nativeimage.api.debug.NIDebuggerService;
+import static org.netbeans.spi.project.ActionProvider.COMMAND_DEBUG;
+import org.netbeans.spi.project.ui.support.MainProjectSensitiveActions;
+import org.netbeans.spi.project.ui.support.ProjectActionPerformer;
+import org.openide.filesystems.FileObject;
+import org.openide.filesystems.FileUtil;
+import org.openide.util.NbBundle;
+
+/**
+ *
+ * @author martin
+ */
+public class DebugNativeImageAction {
+
+    @NbBundle.Messages("CTL_DebugNativeImage=Debug Native Image of {0,choice,-1#Main Project|0#Project|1#Project ({1})|1<{0} Projects}")
+    static Action create() {
```

Comment:
I'd rather connect this with the project support in the future, through a configuration, like in VSCode.
We can not debug any ELF file, but file produced by GraalVM EE native image only.

We can add this once it gets more adopted, and when people realize that there is hardly any reason to create native executables in a different way than using GraalVM native image technology.

Reply:
I wondered about this:

> We can not debug any ELF file, but file produced by GraalVM EE native image only.

So do we really consider integrating a change into the open code base to support a pay only product? It would IMHO be diffferent, if the feature would be in GraalVM CE, but if Oracle wants to keep this commercial, why do we need to support that?

Reply:
The question of _open source purity_ is a fair observation, but possibly a bit too stretched. Open source project do support non-open technologies. Be it access to Gmail, Outlook, Facebook or technologies like mp3 or DVD. 

> We can not debug any ELF file, but file produced by GraalVM EE native image only.

I've just successfully built my [sieve.c](https://github.com/jtulach/sieve/tree/master/c) and managed to debug it via the "GraalVM Native Image" attach:

![obrazek](https://user-images.githubusercontent.com/26887752/113239727-9c0b9600-92ab-11eb-90f5-d5539a3d2db7.png)

There is nothing GraalVM specific in my sample - it's normal C project with a `Makefile`! As I read the code in this pull request, the debugging is ELF general - the only Native Image specific part is mapping of `NIVariable`s. That has only been tested with GraalVM EE, however it should, given a bit of testing and bugfixing, work also with [Quarkus debugging](https://www.youtube.com/watch?v=JqV-NFWupLA) once it becomes widely available.

Reply:
I propose to name this "Native Image Debugger", to be more precise. This will not be commercial feature at the end.

Reply:
O.K. Renamed to Native Image.
In the future, we want to support other native images produced by JVMs, e.g. project Leyden. This infrastructure should make it easy.

Reply:
Thanks for the explanation.